### PR TITLE
PWN-6396 - remove empty balance and change it to Address only with 0 balance

### PR DESCRIPTION
**Pull request by kabramovp2p**
## Jira Ticket

https://p2pvalidator.atlassian.net/browse/PWN-6396
https://p2pvalidator.atlassian.net/browse/PWN-6503
https://p2pvalidator.atlassian.net/browse/PWN-6504

## Description of Work

PWN-6396 - remove empty balance and change it to Address only with 0 balance

### CODE REVIEW

File: app/src/main/java/org/p2p/wallet/send/model/SearchResult.kt [hunk 1]
```diff
@@ -15,8 +18,16 @@ sealed class SearchResult(open val addressState: AddressState) : Parcelable {
     data class AddressOnly constructor(
         override val addressState: AddressState,
         val sourceToken: Token.Active? = null,
-        val date: Date? = null
-    ) : SearchResult(addressState)
+        val date: Date? = null,
+        val balance: Long = EMPTY_BALANCE
+    ) : SearchResult(addressState) {
+        @IgnoredOnParcel
+        val isEmpty = balance == EMPTY_BALANCE
```
Comment (1sakov) [inline app/src/main/java/org/p2p/wallet/send/model/SearchResult.kt:25]: `isEmptyBalance`

File: app/src/main/java/org/p2p/wallet/newsend/ui/search/NewSearchPresenter.kt [hunk 2]
```diff
@@ -89,11 +89,22 @@ class NewSearchPresenter(
...
-        view?.submitSearchResult(result, preselectedToken)
+        launch {
+            val finalResult: SearchResult
+            val preselectedToken: Token.Active?
+            if (result is SearchResult.AddressOnly) {
+                finalResult = result.copyWithBalance(
+                    userInteractor.getBalance(
+                        result.addressState.address.toBase58Instance()
+                    )
+                )
```
Comment (1sakov) [inline app/src/main/java/org/p2p/wallet/newsend/ui/search/NewSearchPresenter.kt:100]:
```suggestion
                val balance = userInteractor.getBalance(result.addressState.address.toBase58Instance()
                finalResult = result.copyWithBalance(balance)
```

**Review by 1sakov — APPROVED** [review]

File: app/src/main/java/org/p2p/wallet/send/interactor/SearchInteractor.kt [hunk 3]
```diff
@@ -94,12 +94,7 @@ class SearchInteractor(
                 )
                 else -> {
                     val balance = userInteractor.getBalance(address)
-                    val hasEmptyBalance = balance == 0L
-                    if (hasEmptyBalance) {
-                        SearchResult.EmptyBalance(addressState, userToken)
-                    } else {
-                        SearchResult.AddressOnly(addressState, userToken)
-                    }
+                    SearchResult.AddressOnly(addressState, userToken, balance = balance)
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/send/interactor/SearchInteractor.kt:97]: named args

File: app/src/main/java/org/p2p/wallet/send/model/SearchResult.kt [hunk 4]
```diff
@@ -15,8 +18,16 @@ sealed class SearchResult(open val addressState: AddressState) : Parcelable {
...
+
+        fun copyWithBalance(balance: Long): AddressOnly {
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/send/model/SearchResult.kt:27]: why just .copy is not enough?

File: app/src/main/java/org/p2p/wallet/send/ui/search/adapter/SearchViewHolder.kt [hunk 5]
```diff
@@ -47,39 +47,49 @@ class SearchViewHolder(
...
 
     private fun renderAddressOnly(item: SearchResult.AddressOnly) {
         with(binding) {
-            if (item.addressState.networkType == NetworkType.BITCOIN) {
-                walletImageView.setImageResource(R.drawable.ic_btc)
-                walletImageView.setPadding(0, 0, 0, 0)
+            val imageIcon = item.sourceToken?.iconUrl
+            val description: String?
+            val imageObject: Any = if (imageIcon != null) {
+                walletImageView.apply {
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/send/ui/search/adapter/SearchViewHolder.kt:71]: no need for apply

File: app/src/main/java/org/p2p/wallet/send/ui/search/adapter/SearchViewHolder.kt [hunk 6]
```diff
@@ -47,39 +47,49 @@ class SearchViewHolder(
...
+                    setPadding(0, 0, 0, 0)
+                }
+                description = getString(
+                    R.string.search_no_other_tokens_description,
+                    item.sourceToken.tokenSymbol
+                )
+                imageIcon
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/send/ui/search/adapter/SearchViewHolder.kt:78]: imageIconUrl

**Review by gslevinkov — APPROVED** [review]

File: app/src/main/java/org/p2p/wallet/send/model/SearchResult.kt (hunk 4 above)
Comment (kabramovp2p) [reply app/src/main/java/org/p2p/wallet/send/model/SearchResult.kt:27]: it is not a simple copy it is copy with overriding balance

**Pull request merged**